module/VuFind/src/VuFind/Db/Service/AuditEventService.php

```diff
@@ -309,12 +312,16 @@ public function purgeEvents(): void
     /**
      * Get the method that called the parent method here.
      *
+     * @param int $intermediateMethods Number of intermediate methods between the actual method adding the event and the
+     * parent method (used for determining the caller name)
+     *
      * @return string
      */
-    protected function getCallerOfParentMethod(): string
+    protected function getCallerOfParentMethod(int $intermediateMethods = 0): string
     {
-        $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 3);
-        $methodParts = [$backtrace[2]['class'] ?? '', $backtrace[2]['function'] ?? ''];
+        $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 3 + $intermediateMethods);
+        $parentPos = 2 + $intermediateMethods;
+        $methodParts = [$backtrace[$parentPos]['class'] ?? '', $backtrace[$parentPos]['function'] ?? ''];
```

Comment:
This is fine as-is, but I think I like it slightly better like this:

```suggestion
        $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 3 + $intermediateMethods);
        $caller = $backtrace[2 + $intermediateMethods];
        $methodParts = [$caller['class'] ?? '', $caller['function'] ?? ''];
```